src/rgw/rgw_lua_utils.cc

```diff
@@ -177,5 +186,34 @@ lua_state_guard::~lua_state_guard() {
   }
 }
 
+void lua_state_guard::runtime_hook(lua_State* L, lua_Debug* ar) {
+  auto now = ceph::real_clock::now();
+  lua_getfield(L, LUA_REGISTRYINDEX, "runtimeguard_max_runtime");
+  auto max_runtime =
+      *static_cast<std::chrono::milliseconds*>(lua_touserdata(L, -1));
+  lua_getfield(L, LUA_REGISTRYINDEX, "runtimeguard_start_time");
+  auto start_time =
+      *static_cast<ceph::real_clock::time_point*>(lua_touserdata(L, -1));
+  lua_pop(L, 2);
+  auto elapsed =
+      std::chrono::duration_cast<std::chrono::milliseconds>(now - start_time);
+
+  if (elapsed > max_runtime) {
+    luaL_error(L, "Lua runtime limit exceeded");
+  }
+}
+
+void lua_state_guard::set_runtime_hook() {
+  lua_pushlightuserdata(state,
+                        const_cast<std::chrono::milliseconds*>(&max_runtime));
+  lua_setfield(state, LUA_REGISTRYINDEX, "runtimeguard_max_runtime");
```

Comment:
better to use `static const char*` for the field names, since they are used in 2 places.
this will guarantee there is no mismatch